on_node_configured.sh

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/env bash
+
+# https://gist.github.com/mohanpedala/1e2ff5661761d3abd0385e8223e16425#file-bash_strict_mode-md
+set -euo pipefail
+
+#####
+# Script arguments
+#####
+SLURM_JWT_KEY="$1"
+# GitHub release tag for UQLE CLI tool
+# CLI_TAG="$2"
+# GitHub OAuth token - should have read access to UQLE CLI releases, and the UQLE stack repository
+# MACHINE_USER_TOKEN="$3"
+# UQLE_API_HOST="$4"
+
+# global variables
+JWT_KEY_DIR=/var/spool/slurm.state
+JWT_KEY_FILE="$JWT_KEY_DIR"/jwt_hs256.key
+SLURMDBD_USER="slurm"
+SLURM_PASSWORD_FILE=/root/slurmdb.password
+
+. /etc/parallelcluster/cfnconfig
+
+echo "Node type: $cfn_node_type"
+
+function modify_slurm_conf() {
+    # add JWT auth and accounting config to slurm.conf
+    # /opt/slurm is shared via nfs, so this only needs to be configured on head node
+
+    cat >>/opt/slurm/etc/slurm.conf <<EOF
+# Enable jwt auth for Slurmrestd
+AuthAltTypes=auth/jwt
+# ACCOUNTING
+JobAcctGatherType=jobacct_gather/linux
+AccountingStorageType=accounting_storage/slurmdbd
+AccountingStorageHost=localhost
+AccountingStoragePort=6819
+EOF
+}
+
+function configure_users_common() {
+    sysctl user.max_user_namespaces=15000
+    usermod --add-subuids 165536-231071 --add-subgids 165536-231071 slurm
+}
+
+function configure_users_head_node() {
+    configure_users_common
+
+    cat <<'EOF' | tee -a /home/centos/.bashrc /home/slurm/.bashrc
```

Comment:
*non-blocking, query:* I'm slightly confused by the syntax here - why is the EOF in single quotes? Where does the EOF start here? Looks like some bash dark magic

Reply:
See [here](https://www.gnu.org/software/bash/manual/bash.html#Here-Documents)

> If any part of word is quoted, the delimiter is the result of quote removal on word, and the lines in the here-document are not expanded

Basically, using quotes means that variables are not expanded. This is desired here, as we want the literal text to be appended to the file.

I have just noticed you can also add a `-` symbol and keep natural indentation! I won't do that on this MR though.

Reply:
Every time I learn something new about bash it just becomes more incomprehensible. This is cool, though!